contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -992,26 +992,42 @@ class tsqlCommonMutator : public TSqlParserBaseListener
         /* rewrite CROSS/OUTER APPLY to CROSS/LEFT JOIN LATERAL */
         if (ctx->APPLY())
         {
+            /* In PG, LATERAL joins can only be applied to table_refs and not bare table calls, so check for that case.
+             * Luckily, in PG all LATERAL joins (cross and left) require that the joined table_ref has an alias, so just
+             * check for any alias clause to determine whether to use a lateral join or just regular join. The only other
+             * caveat is with with function calls (which don't need an alias in either SQL Server or PG), so also check for that case.
+             */
+            bool lateral;
+            if (ctx->table_source_item(1) && (ctx->table_source_item(1)->function_call() || ctx->table_source_item(1)->as_table_alias(0)))
+                lateral = true;
+            else
+                lateral = false;
+            
             if (ctx->CROSS())
             {
                 rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
-                                                                std::make_pair(ctx->APPLY()->getText(), "JOIN LATERAL")));
+                                                                std::make_pair(ctx->APPLY()->getText(), lateral ? "JOIN LATERAL" : "JOIN")));
             }
             else if (ctx->OUTER())
             {
                 /* replace OUTER with LEFT */
                 rewritten_query_fragment.emplace(std::make_pair(ctx->OUTER()->getSymbol()->getStartIndex(),
                                                                 std::make_pair(ctx->OUTER()->getText(), "LEFT")));
-                rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
-                                                                std::make_pair(ctx->APPLY()->getText(), "JOIN LATERAL")));
+
+                if (lateral)
+                {
+                    rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
+                                                                    std::make_pair(ctx->APPLY()->getText(), "JOIN LATERAL")));
+                }
+                else
+                {
+                    rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
+                                                                    std::make_pair(ctx->APPLY()->getText(), "JOIN")));
+                }
```

Comment:
Why not using `lateral ? "JOIN LATERAL" : "JOIN"`?

Reply:
good point, I'll use that